### ISSUE TITLE
Provide a way to compress/decompress messages going through the proxy

### ISSUE THREAD

**Issue by johanrhodin** [cloudamqp/amqproxy#168]
Publishing Client -> Proxy (compress) -> Broker -> Proxy (decompress) -> Consuming client.

Could also (potentially more complicated) provide a way to encrypt/decrypt message bodies.

**Comment by spuun** [comment]
I think body encryption may be pretty simple and straight forward.

Dunno how compression would work. Your thinking about compressing all network traffic, not just part of the frames, right? That would require compression and decompression in the broker too.

**Comment by carlhoerberg** [comment]
Would come with a lot of cavets. Like adding Content-Transfer header etc to know which bodies to decompress and which to decrypt. What about messages from other sources? What about clients that want to consume messages published with compression/encryption?

**Comment by spuun** [comment]
Right. All traffic would have to go through amqproxy, and the same instance (or they must have the same keep). And key rotation is a problem. So, adding the code to encrypt/decrypt is straight forward, but to do a reasonable administration is not.
